src/dir.rs

```diff
@@ -0,0 +1,198 @@
+use {Error, Result};
+use errno::Errno;
+use libc;
+use std::os::unix::io::{AsRawFd, IntoRawFd, RawFd};
+use std::{ffi, fmt, ptr};
+
+#[cfg(target_os = "linux")]
+use libc::{dirent64 as dirent, readdir64_r as readdir_r};
+
+#[cfg(not(target_os = "linux"))]
+use libc::{dirent, readdir_r};
+
+/// An open directory.
+///! This is a lower-level interface than `std::fs::ReadDir`. Notable differences:
```

Comment:
Does this render correctly in Rust Doc?  Normally `///!` annotates the enclosing object, which in this case would be the file.  To describe `Dir` instead, you should have all `///`s, with a blank line after the first.

Reply:
It did, but fixed anyway. Also adding some extra whitespace to the docstrings so the one-line extract would be less overwhelming.